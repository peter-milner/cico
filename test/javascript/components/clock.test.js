import React from 'react'
import { act } from 'react-dom/test-utils'
import { mount } from 'enzyme'

import Clock from '../../../app/javascript/components/clock'

const render = () => {
    return mount(
        <Clock />
    )
}

describe('<Clock />', () => {
    test('snapshot', () => {
        const rendered = render()
        expect(rendered).toMatchSnapshot()
    })

    test('clock is ticking', async () => {
        const rendered = render()
        const time = new Date().toLocaleTimeString()
        expect(rendered.text()).toEqual(time)
        await act(async () => {
            await new Promise((r) => setTimeout(r, 1000))
        })
        rendered.update()
        expect(rendered.text()).not.toEqual(time)     
    })
})
