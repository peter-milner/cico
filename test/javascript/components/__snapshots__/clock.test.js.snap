// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock /> snapshot 1`] = `ReactWrapper {}`;
